src/test/java/com/faforever/client/mod/ModDetailControllerTest.java

```diff
@@ -157,6 +161,29 @@ public void testSetModWithNoUploader() {
     verify(modService).loadThumbnail(modVersion);
   }
 
+  @Test
+  public void testWithRepositoryURL() throws MalformedURLException {
+    modVersion = Instancio.of(ModVersion.class)
+                                     .set(field(Mod::repositoryURL),
+                                          URI.create("https://github.com/FAForever/fa").toURL())
+                                     .create();
+    runOnFxThreadAndWait(() -> instance.setModVersion(modVersion));
+    WaitForAsyncUtils.waitForFxEvents();
```

Comment:
technically you don't need WaitForAsyncUtils.waitForFxEvents if you run things with runOnFxThreadAndWait

Reply:
Processed in https://github.com/FAForever/downlords-faf-client/pull/3192/commits/23ab26f875ea07ac10fe75181bc01b1b4813ba03